pydantic/typing.py

```diff
@@ -354,10 +360,7 @@ def is_none_type(type_: Any) -> bool:
 else:
 
     def is_none_type(type_: Any) -> bool:
-        for none_type in NONE_TYPES:
-            if type_ is none_type:
-                return True
-        return False
+        return type_ in NONE_TYPES
```

Comment:
This looks weird enough that we might have done it for some strange reason, can we confirm this was just a code smell?

Reply:
The existing approach(checking with `is`) does not work in  `Python 3.9` and `typing-extension==4.6.0`. That's why I've changed it.

Note: the new approach is also Ok with `typing-extension==4.5.0`.